realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.java

```diff
@@ -243,13 +243,14 @@ private void emitCopyToRealmMethod(JavaWriter writer) throws IOException {
                 EnumSet.of(Modifier.PUBLIC),
                 "Realm", "realm", "E", "obj", "boolean", "update", "Map<RealmObject, RealmObjectProxy>",  "cache"
         );
-
-        writer.emitStatement("Class<E> clazz = (Class<E>) ((obj instanceof RealmObjectProxy) ? obj.getClass().getSuperclass() : obj.getClass())");
+        writer.emitSingleLineComment("This cast is correct because obj is either ");
```

Comment:
Cool, hadn't noticed you could do this with JavaWriter. We should probably use this a bit more.